pkg/kubectl/cmd/util/helpers.go

```diff
@@ -155,6 +156,8 @@ func checkErr(err error, handleErr func(string, int)) {
 			default:
 				handleErr(fmt.Sprintf("the server doesn't have a resource type %q", err.PartialResource.Resource), DefaultErrorExitCode)
 			}
+		case *discovery.ErrParseServerVersionFailed:
+			handleErr(fmt.Sprintf("error retrieving server version: %s", err.Error()), DefaultErrorExitCode)
```

Comment:
Still the same question: why do we need special error handling in kubectl? We better report a good error message from the discovery code and use the default error path here.

Reply:
maybe we need different error message between client-go and kubectl. error message in kubectl giving the message what error and in client-go (discovery) giving a reason why it's an error.

of course, if this is not necessary i will update it.